src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -329,6 +330,169 @@ private static BoundCall CreateParameterlessCall(CSharpSyntaxNode syntax, BoundE
             { WasCompilerGenerated = true };
         }
 
+        // A synthesized entrypoint that forwards all calls to an async Main Method
+        internal sealed class AsyncForwardEntryPoint : SynthesizedEntryPointSymbol
+        {
+            // The user-defined asyncrhonous main method.
+            private MethodSymbol _userMain;
+            private DiagnosticBag _diagnosticBag;
+            private CSharpCompilation _compilation;
```

Comment:
Why is this stored as a field?  It seems to only be used in the constructor of the type.